### ISSUE TITLE
No newline at the end of blocks, similar to `vue/multiline-html-element-content-newline`

### ISSUE THREAD

**Issue by seahindeniz** [vuejs/eslint-plugin-vue#2086]
**Please describe what the rule should do:**

Files can have an empty line at the end of a block. There can be a new rule that could work like the `vue/multiline-html-element-content-newline` which expects you to remove the empty newline before closing the `template`

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
Bad
```vue
<script setup>
import { ref } from 'vue';

const foo = ref(null);

</script>

<style>
.foo {}

</style>
```
Good
```vue
<script setup>
import { ref } from 'vue';

const foo = ref(null);
</script>

<style>
.foo {}
</style>
```
**Additional context**

IMO, having one rule to support all kinds of blocks is better than having a separate rule for all block types.
Something called `vue/end-of-block-newline` with the following options, perhaps:
- second parameter: `'always' | 'never'` default `'never'`
- third parameter:
  - ignore: `Array<('script[setup]' | 'template' | 'script:not([setup])' | 'style')>` default `[ ]`

**Comment by ota-meshi** [comment]
Thank you for posting the issue.
But I think the rule already exist for that.
https://eslint.vuejs.org/rules/block-tag-newline.html

**Comment by seahindeniz** [comment]
OMG, I'm so sorry 😨 I have searched for this rule here, on repo and the website and couldn't find anything that resembles. Thank you for pointing out the rule, it satisfies
